src/realm-settings/KeysProvidersTab.tsx

```diff
@@ -317,22 +349,24 @@ export const KeysTabInner = ({ components, refresh }: KeysTabInnerProps) => {
           {(filteredComponents.length === 0
             ? components
             : filteredComponents
-          ).map((component: ComponentData, idx) => (
+          ).map((component: ComponentData, idx: number) => (
             <DataListItem
               draggable
               aria-labelledby={"aria"}
-              key={`data${idx}`}
-              id={`data${idx}`}
+              key={component.id}
+              id={component.id}
             >
               <DataListItemRow key={idx} data-testid={"data-list-row"}>
```

Comment:
The `key` prop here is not needed as these elements are static.
```suggestion
              <DataListItemRow data-testid={"data-list-row"}>
```